pyswell/pyswell.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/python3
+
+ASSET_ID = ""
+
+# Default list. Feel free to edit.
+NEVER_COLLECT  = ["\.7z$", "\.-wal$", "\.accdb$", "\.accde$", "\.accdr$", "\.accdt$", "\.accdu$", "\.asl$", "\.bin$", "\.csv$", "\.dat$", "\.db-shm$", "\.db$", "\.doc$", "\.docb$", "\.docm$", "\.docx$", "\.dot$", "\.dotm$", "\.DS_Store$", "\.emlx$", "\.img$", "\.json$", "\.key$", "\.md$", "\.mdb$", "\.mpp$", "\.msf$", "\.msg$", "\.numbers$", "\.odb$", "\.odp$", "\.ods$", "\.odt$", "\.one$", "\.oft$", "\.ost$", "\.otf$", "\.pages$", "\.pdf$", "\.plist$", "\.pot$", "\.potx$", "\.ppam$", "\.pps$", "\.ppsm$", "\.ppsx$", "\.ppt$", "\.pptm$", "\.pptx$", "\.pst$", "\.pub$", "\.rar$", "\.rtf$", "\.sldm$", "\.sldx$", "\.sqlite-journal$", "\.sqlite$", "\.swp$", "\.ttf$", "\.tracev3$", "\.txt$", "\.vcf$", "\.xla$", "\.xlam$", "\.xlm$", "\.xls$", "\.xlsb$", "\.xlsm$", "\.xlsx$", "\.xlt$", "\.xltm$", "\.xltx$", "\.xlw$", "\.wpd$", "\.xps$", "\.zip"]
+ALWAYS_COLLECT = ["\.app$", "\.appx$", "\.appxbundle$", "\.bat$", "\.class$", "\.cmd$", "\.com$", "\.crx$", "\.dll$", "\.dmg$", "\.drv$", "\.dylib$", "\.ear$", "\.efi$", "\.elf$", "\.exe$", "\.hta$", "\.iso$", "\.jar$", "\.java$", "\.js$", "\.lib$", "\.lnk$", "\.msi$", "\.nar$", "\.pkg$", "\.pl$", "\.ps1$", "\.py$", "\.pyc$", "\.rb$", "\.scr$", "\.sct$", "\.sfx$", "\.sh$", "\.so$", "\.sys$", "\.vb$", "\.vba$", "\.vbs$", "\.vbscript$", "\.war$", "\.xpi$", "\.zsh"]
+
+import os          # Used for crawling directories
+import json        # JSON payloads
+import subprocess  # Required to call 'file' command
+import re          # Regex for extension matching
+import requests    # Calling REST API
+from requests.exceptions import ConnectTimeout
+import sys         # Reading CLI arguments
+import hashlib     # Calculating SHA256
+
+# Help screen if no args
+if (len(sys.argv)) == 1:
```

Comment:
Let's use something like argparse instead. I made a plug and play example here: 

```
#!/usr/bin/env python

import argparse
import sys
import os


def main():
    # create the top-level parser
    parser = argparse.ArgumentParser(prog='pySwell')
    subparsers = parser.add_subparsers(help='available commands')

    # By default, sub-parseres are mutually exclusive
    # Create parser for mkasset
    parser_mkasset = subparsers.add_parser('mkasset')
    parser_mkasset.add_argument('--name',
                                required=True,
                                help="Name of the asset to create")
    parser_mkasset.add_argument('--env_id',
                                required=True,
                                help="Environment ID to create the asset in")
    parser_mkasset.add_argument('--api_key',
                                required=True,
                                help="API key to create the asset with")
    parser_mkasset.set_defaults(func=make_asset)

    # Create parser for upload command
    parser_upload = subparsers.add_parser('upload')
    parser_upload.add_argument('--path',
                                   required=True,
                                   help="Path to the directory to upload")
    parser_upload.set_defaults(func=upload)

    # If no arguments have been passed, print help
    if len(sys.argv) == 1:
        parser.print_help()

    asset_id = os.getenv("ASSET_ID")
    if asset_id is None:
        sys.exit("The ASSET_ID environment variable is not set, exiting...")

    args = parser.parse_args()
    args.func(args, asset_id)


def make_asset(args, asset_id):
    # Put all make asset code here
    print(args.name)
    print(args.env_id)
    print(args.api_key)


def upload(args, asset_id):
    # put all upload code here
    print(args.path)


if __name__ == "__main__":
    main()
```

Reply:
Done